core/src/gossip/orderbook.rs

```diff
@@ -333,9 +333,10 @@ impl GossipProtocolExecutor {
             .map_err(|err| GossipError::StarknetRequest(err.to_string()))?
         {
             log::info!("got order with invalid merkle root, skipping...");
-            return Err(GossipError::ValidCommitmentVerification(
-                "invalid merkle root, not in contract history".to_string(),
-            ));
+            // TODO: Once the contract implements foreign field Merkle, error on this test
```

Comment:
This blocks MPCs from happening correctly, we will re-enable this check once the Cairo 1.0 rewrite supports `Dalek` scalars as inputs/outputs to/from the Merkle tree